plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeSplitManager.java

```diff
@@ -184,9 +184,9 @@ private Stream<DeltaLakeSplit> getSplits(
                                 .filter(column -> partitionValues.containsKey(((DeltaLakeColumnHandle) column).getName()))
                                 .collect(toImmutableMap(identity(), column -> {
                                     DeltaLakeColumnHandle deltaLakeColumn = (DeltaLakeColumnHandle) column;
-                                    return NullableValue.of(
+                                    return new NullableValue(
                                             deltaLakeColumn.getType(),
-                                            deserializePartitionValue(deltaLakeColumn, addAction.getCanonicalPartitionValues().get(deltaLakeColumn.getName())));
+                                            deserializePartitionValue(deltaLakeColumn, partitionValues.get(deltaLakeColumn.getName())));
```

Comment:
nit: for a minimal change you could keep `addAction.getCanonicalPartitionValues()` here 
(or make a separate commit like in https://github.com/trinodb/trino/pull/12052: https://github.com/trinodb/trino/pull/12052/commits/a03b205450530cb9741e5176cc8b1eec65d8cbe4)

Reply:
ok I will remove it. I changed it because it hurt my eyes..